mpi/vcpu.c

```diff
@@ -0,0 +1,109 @@
+/************************************************************\
+ * Copyright 2024 Lawrence Livermore National Security, LLC
+ * (c.f. AUTHORS, NOTICE.LLNS, COPYING)
+ *
+ * This file is part of the Flux resource manager framework.
+ * For details, see https://github.com/flux-framework.
+ *
+ * SPDX-License-Identifier: LGPL-3.0
+\************************************************************/
+
+#include <mpi.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <assert.h>
+#include <sched.h>
+#define MASTER 0
+
+int sched_getcpu (void);
```

Comment:
Do you mind if I leave this? Both cray's compilers and gcc throw a warning if I pull it out:

```
(s=33,d=0) tioga11 ~/flux-test-collective/mpi (vcpu-test)$ mpicc -o vcpu vcpu.c
vcpu.c: In function 'main':
vcpu.c:65:11: warning: implicit declaration of function 'sched_getcpu'; did you mean 'sched_getparam'? [-Wimplicit-function-declaration]
   65 |     cpu = sched_getcpu ();
      |           ^~~~~~~~~~~~
      |           sched_getparam
```

Reply:
You get this warning because you are not defining `_GNU_SOURCE` before including `sched.h` as documented in the "Feature Test Macro Requirements" I referenced above.

It is slightly bad practice to define a prototype for an external function if there exists a valid header for it. In the case the prototype is incorrect you may have a runtime crash which would have been easily avoided by including the correct prototype.

Reply:
Thanks. I just recalled that @garlick had asked for a small blurb at the top of the test describing its pass/fail conditions. Added that too.